replica/table.cc

```diff
@@ -1149,6 +1150,7 @@ compaction_group::update_sstable_lists_on_off_strategy_completion(sstables::comp
         virtual void execute() override {
             _cg.set_main_sstables(std::move(_new_main_list));
             _cg.set_maintenance_sstables(std::move(_new_maintenance_list));
+            // FIXME: the following is not exception safe
             _t.refresh_compound_sstable_set();
             // Input sstables aren't not removed from backlog tracker because they come from the maintenance set.
             _cg.backlog_tracker_adjust_charges({}, _new_main);
```

Comment:
interesting, if you recall, we wanted to stop maintaining a clone of sstable set in backlog tracker. and turns out we also suffer with exception unsafety as sstable set and backlog tracker are updated independently. perhaps we could implement a backlog_tracked_sstable_set later, so we kill two birds with one stone.

Reply:
But the backlog tracker is per compaction group, while the compound sstable set covers the whole table.

Reply:
Please review the updated series, in particular the last patches in the series:
- sstable_set: keep impl as shared_ptr
- sstable_set: keep sstable_set_impl_ptr in incremental selectors impl
- sstable_set: allow updating in-place
- sstable_set: compound_sstable_set: point to underlying sstable_set:s

They allow the compound_sstable_set to safely keep pointers to the underlying sstable sets, and allow updating those in-place by replacing the shared_ptr<sstable_set_impl> they keep when updating the cache.

With that, `table::refresh_compound_sstable_set` is no longer required.